js/components/controlButton.js

```diff
@@ -0,0 +1,316 @@
+var React = require('react');
+var classNames = require('classnames');
+var AccessibleButton = require('./accessibleButton');
+var Icon = require('./icon');
+var Tooltip = require('./tooltip');
+var Utils = require('./utils');
+var CONSTANTS = require('../constants/constants');
+
+/**
+ * Template component that is used for buttons that don't have accessibility enabled.
+ * The component is picked at runtime by ControlButton depending on the props that
+ * were passed to it.
+ * @param {Object} props Component's props
+ * @return {Component} React component
+ */
+var NonAccessibleButton = function(props) {
+  return (
+    <a
+      {...props}
+      aria-hidden={props.ariaHidden}>
+      {props.children}
+    </a>
+  );
+};
+
+/**
+ * Component used for action buttons within the skin. Currently used for both control
+ * bar and skip buttons. Implements common functionality for these buttons which includes
+ * icons, highlighting, tooltips and accessibility.
+ */
+var ControlButton = React.createClass({
+  /**
+   * Stores a ref to this component's main element or component.
+   * @private
+   * @param {*} buttonRef Either a reference to an AccessibleButton or a DOM element,
+   * depending on the component that was rendered
+   */
+  storeRef: function(buttonRef) {
+    this.buttonRef = buttonRef;
+    // Pass ref on to parent, if subscribed
+    if (typeof this.props.onRef === 'function') {
+      this.props.onRef(this.buttonRef);
+    }
+  },
+
+  /**
+   * Chooses a component (either AccessibleButton or NonAccessibleButton) to use
+   * as base for the ControlButton depending on whether or not accessibility is
+   * enabled for this particular instance. Accessibility is enabled by default
+   * unless the ariaHidden prop is set to false explicitly.
+   * @private
+   * @return {Component} An AccessibleButton or NonAccessibleButton component, depending on the case
+   */
+  getButtonComponent: function() {
+    var Component;
```

Comment:
Is it standard to capitalize component when it is used like this?

Reply:
Yes, since this is a reference that is used directly in the JSX, where components are usually capitalized.